main/src/hadd.cxx

```diff
@@ -1,39 +1,47 @@
-  The target file is newly created and must not be
-  identical to one of the source files.
+  \brief This program will add histograms (see note) and Trees from a list of root files and write them to a target root
+  file. The target file is newly created and must not be identical to one of the source files.
 
   Syntax:
   ```{.cpp}
-       hadd targetfile source1 source2 ...
+       hadd [flags] targetfile source1 source2 ...
   ```
-  or
-  ```{.cpp}
-       hadd -f targetfile source1 source2 ...
-  ```
-  (targetfile is overwritten if it exists)
+
+  Flags can be passed before or after the positional arguments.
+  The first positional (non-flag) argument will be interpreted as the targetfile.
+  After that, the first sequence of positional arguments will be interpreted as the input files.
+  If two sequences of positional arguments are separated by flags, all sequences but the first
+  will be ignored.
+  If a flag requires an argument, the argument can be specified in any of these ways:
+
+     # All equally valid:
+     -j 16
+     -j16
+     -j=16
+
+   The first syntax is the preferred one since it's backward-compatible with previous versions of hadd.
+   The -f flag is an exception to this rule: it only supports the `-f[0-9]` syntax.
+
+   Note that merging multiple flags is NOT supported: `-jfa` will be interpreted as -j=fa, which is invalid!
+
+   The flags are as follows:
 
   \param -a   Append to the output
+  \param -cachesize Resize the prefetching cache use to speed up I/O operations (use 0 to disable).
+  \param -d   Carry out the partial multiprocess execution in the specified directory
+  \param -dbg Enable verbosity. If -j was specified, do not not delete partial files stored inside working directory.
+  \param -experimental-io-features `<feature>` Enables the corresponding experimental feature for output trees. \see ROOT::Experimental::EIOFeatures
   \param -f   Force overwriting of output file.
-  \param -f[0-9] Set target compression level. 0 = uncompressed, 9 = highly compressed. Default is 1 (kDefaultZLIB).
-                 You can also specify the full compresion algorithm, e.g. -f206
-  \param -fk  Sets the target file to contain the baskets with the same compression
-              as the input files (unless -O is specified). Compresses the meta data
-              using the compression level specified in the first input or the
-              compression setting after fk (for example 206 when using -fk206)
+  \param -f[0-9] Set target compression level. 0 = uncompressed, 9 = highly compressed. Default is 1 (kDefaultZLIB). You can also specify the full compression algorithm, e.g. -f206
+  \param -fk  Sets the target file to contain the baskets with the same compression as the input files (unless -O is specified). Compresses the meta data using the compression level specified in the first input or the compression setting after fk (for example 206 when using -fk206)
```

Comment:
> (for example 206 when using -fk206)

A nitpick, but here and in the `-f` parameter it would be nice to show `505` rather than `206` as we know it's a more widely sensible compression setting. Not that this comment is suggesting to use 206 per se, but seeing that value in the manual of the CLI might lead some people to just use that to try it out and then start using it indiscriminately